### PR TITLE
Migrate terraform-aws-cloudwatch-to-splunk to Terraform 0.12

### PR DESCRIPTION
**Pull request by JonRoma**

### CODE REVIEW

File: iam.tf [hunk 1]
```diff
@@ -55,27 +60,30 @@ data "aws_iam_policy_document" "assume-role-policy" {
     actions = ["sts:AssumeRole"]
 
     principals {
-      type        = "AWS"
-      identifiers = ["${format("arn:aws:iam::%s:root", data.aws_caller_identity.current.account_id)}"]
+      type = "AWS"
+      identifiers = [format(
```
Comment (ddriddle) [inline iam.tf:64]: This is short enough to be a one liner. IMO.

File: main.tf [hunk 2]
```diff
@@ -13,21 +13,21 @@ resource "aws_lambda_function" "default" {
...
-  timeout     = "${var.timeout}"
+  runtime     = var.runtime
+  memory_size = var.memory_size
+  timeout     = var.timeout
   handler     = "index.handler"
-  publish     = "true"
-  role        = "${aws_iam_role.default.arn}"
+  publish     = true
+  role        = aws_iam_role.default.arn
   s3_bucket   = "drone-${local.region}-${local.account_id}"
```
Comment (ddriddle) [inline main.tf:24]: We need to do something about this hardcoded bucket.

File: main.tf [hunk 3]
```diff
@@ -13,21 +13,21 @@ resource "aws_lambda_function" "default" {
...
   s3_key      = "splunk-aws-serverless-apps/splunk-cloudwatch-logs-processor.zip"
```
Comment (ddriddle) [inline main.tf:25]: We need to do something about this hardcoded s3 key.

**Review by ddriddle — CHANGES_REQUESTED** [review]

File: iam.tf (hunk 1 above)
Comment (JonRoma) [reply iam.tf:64]: This doesn't really fit on one line nicely, so I think it is probably more readable as is.

File: main.tf (hunk 2 above)
Comment (JonRoma) [reply main.tf:24]: I filed an issue; see #3.

File: main.tf (hunk 3 above)
Comment (JonRoma) [reply main.tf:25]: Again, this should be reviewed. This module was created  to deploy a single lambda function, but ought to be made more generic if possible. The S3 key being hardcoded would preclude this. I've added this to #3.

Comment (JonRoma) [reply main.tf:25]: Let's do this in the same card where we refactor `terraform-aws-cloudfront-directory-index`. They are both related, and if we can show that they work cleanly, I am more than happy to eliminate the S3 bucket once it becomes unnecessary.

**Review by ddriddle — APPROVED** [review]

**Pull request merged**